diff --git a/src/AppBundle/Security/WeirdFormAuthenticator.php b/src/AppBundle/Security/WeirdFormAuthenticator.php
index c5975f5..d9cc842 100644
--- a/src/AppBundle/Security/WeirdFormAuthenticator.php
+++ b/src/AppBundle/Security/WeirdFormAuthenticator.php
@@ -10,6 +10,7 @@ use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\RouterInterface;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\CustomUserMessageAuthenticationException;
 use Symfony\Component\Security\Core\Security;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
@@ -46,7 +47,9 @@ class WeirdFormAuthenticator extends AbstractGuardAuthenticator
 
         // this looks like a weird username
         if (substr($username, 0, 1) == '@') {
-            return;
+            throw new CustomUserMessageAuthenticationException(
+                'Starting a username with @ is weird, don\'t you think?'
+            );
         }
 
         return $this->em->getRepository('AppBundle:User')
@@ -60,11 +63,15 @@ class WeirdFormAuthenticator extends AbstractGuardAuthenticator
         }
 
         if ($credentials['answer'] != 42) {
-            return;
+            throw new CustomUserMessageAuthenticationException(
+                'Don\'t you read any books?'
+            );
         }
 
         if (!$credentials['terms']) {
-            return;
+            throw new CustomUserMessageAuthenticationException(
+                'Agree to our terms!!!'
+            );
         }
 
         return true;
